2025-04-02 12:22:51,469 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-02 12:22:51,470 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-02 12:22:51,470 - INFO - Geocoder initialized successfully
2025-04-04 12:02:46,212 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:02:46,213 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:02:46,213 - INFO - Geocoder initialized successfully
2025-04-04 12:02:46,225 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=chennai%2C+Maharashtra&format=json&limit=1
2025-04-04 12:02:46,226 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 12:02:46,700 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=chennai%2C+Maharashtra&format=json&limit=1 HTTP/1.1" 200 529
2025-04-04 12:02:46,701 - DEBUG - RateLimiter sleep(0.5234146000002511)
2025-04-04 12:02:47,226 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 12:02:48,027 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/1.1" 200 441
2025-04-04 12:02:48,029 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 12:02:48,708 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/1.1" 200 None
2025-04-04 12:05:57,605 - WARNING - Not Found: /supplier
2025-04-04 12:08:05,024 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:08:05,024 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:08:05,025 - INFO - Geocoder initialized successfully
2025-04-04 12:08:05,025 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=chennai%2C+Tamil+Nadu&format=json&limit=1
2025-04-04 12:08:05,026 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 12:08:06,055 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=chennai%2C+Tamil+Nadu&format=json&limit=1 HTTP/1.1" 200 418
2025-04-04 12:08:06,056 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 12:08:07,805 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/1.1" 200 441
2025-04-04 12:08:07,808 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 12:08:08,510 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/1.1" 200 None
2025-04-04 12:08:37,990 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:08:37,991 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 12:08:37,991 - INFO - Geocoder initialized successfully
2025-04-04 12:08:37,991 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=chennai%2C+Tamil+Nadu&format=json&limit=1
2025-04-04 12:08:37,992 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 12:08:38,564 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=chennai%2C+Tamil+Nadu&format=json&limit=1 HTTP/1.1" 200 418
2025-04-04 12:08:38,564 - DEBUG - RateLimiter sleep(0.4271254999912344)
2025-04-04 12:08:38,992 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 12:08:39,775 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/1.1" 200 441
2025-04-04 12:30:40,779 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:31:26,751 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:31:49,902 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\urls.py changed, reloading.
2025-04-04 12:37:02,297 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:37:20,725 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\urls.py changed, reloading.
2025-04-04 12:45:45,547 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:46:08,757 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:55:34,946 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:57:07,758 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:57:18,343 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 12:59:31,800 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\urls.py changed, reloading.
2025-04-04 12:59:42,798 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\urls.py changed, reloading.
2025-04-04 13:23:56,554 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 13:23:56,555 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 13:23:56,557 - WARNING - Bad Request: /manufacturer/calculate-carbon-footprint/
2025-04-04 13:45:04,654 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\manufacturer\views.py changed, reloading.
2025-04-04 13:48:25,197 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 13:48:25,198 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 13:48:25,198 - INFO - Geocoder initialized successfully
2025-04-04 13:48:25,198 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=chennai%2C+Tamil+Nadu&format=json&limit=1
2025-04-04 13:48:25,199 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 13:48:26,020 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=chennai%2C+Tamil+Nadu&format=json&limit=1 HTTP/1.1" 200 418
2025-04-04 13:48:26,021 - DEBUG - RateLimiter sleep(0.17771499999798834)
2025-04-04 13:48:26,199 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 13:48:27,073 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/1.1" 200 441
2025-04-04 13:48:27,075 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 13:48:28,117 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/1.1" 200 None
2025-04-04 19:39:37,610 - WARNING - Not Found: /favicon.ico
2025-04-04 19:41:08,536 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 19:41:08,537 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 19:41:08,537 - INFO - Geocoder initialized successfully
2025-04-04 19:41:08,537 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Phoenix%2C+Arizona&format=json&limit=1
2025-04-04 19:41:08,539 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 19:41:10,173 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Phoenix%2C+Arizona&format=json&limit=1 HTTP/11" 200 451
2025-04-04 19:41:10,175 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 19:41:11,341 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/11" 200 441
2025-04-04 19:41:11,342 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 19:41:12,332 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/11" 400 289
2025-04-04 19:41:12,333 - WARNING - Road route error: {"error":{"code":2004,"message":"Request parameters exceed the server configuration limits. The approximated route distance must not be greater than 6000000.0 meters."},"info":{"engine":{"build_date":"2025-03-14T11:07:03Z","graph_version":"1","version":"9.1.1"},"timestamp":1743775873097}}
2025-04-04 19:41:12,334 - WARNING - Bad Request: /supplier/calculate-route/
2025-04-04 19:41:23,214 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 19:41:23,214 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 19:41:23,214 - INFO - Geocoder initialized successfully
2025-04-04 19:41:23,214 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Phoenix%2C+Arizona&format=json&limit=1
2025-04-04 19:41:23,215 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 19:41:24,004 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Phoenix%2C+Arizona&format=json&limit=1 HTTP/11" 200 451
2025-04-04 19:41:24,005 - DEBUG - RateLimiter sleep(0.2097451000008732)
2025-04-04 19:41:24,216 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Mumbai%2C+Maharashtra&format=json&limit=1
2025-04-04 19:41:25,314 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Mumbai%2C+Maharashtra&format=json&limit=1 HTTP/11" 200 441
2025-04-04 19:41:25,315 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 19:41:26,497 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/11" 400 289
2025-04-04 19:41:26,498 - WARNING - Road route error: {"error":{"code":2004,"message":"Request parameters exceed the server configuration limits. The approximated route distance must not be greater than 6000000.0 meters."},"info":{"engine":{"build_date":"2025-03-14T11:07:03Z","graph_version":"1","version":"9.1.1"},"timestamp":1743775887262}}
2025-04-04 19:41:26,499 - WARNING - Bad Request: /supplier/calculate-route/
2025-04-04 19:48:14,898 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\manufacturer\forms.py changed, reloading.
2025-04-04 19:48:20,140 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\manufacturer\forms.py changed, reloading.
2025-04-04 19:48:50,695 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\forms.py changed, reloading.
2025-04-04 19:51:22,763 - ERROR - Internal Server Error: /supplier/register/
Traceback (most recent call last):
  File "C:\Users\Ganesh\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ganesh\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\views.py", line 46, in supplier_register
    first_name=form.cleaned_data['first_name'],
               ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: 'first_name'
2025-04-04 19:51:55,717 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\forms.py changed, reloading.
2025-04-04 19:52:01,046 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\manufacturer\forms.py changed, reloading.
2025-04-04 20:13:38,662 - WARNING - Not Found: /favicon.ico
2025-04-04 20:25:31,834 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:25:31,834 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:25:31,835 - INFO - Geocoder initialized successfully
2025-04-04 20:25:31,835 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1
2025-04-04 20:25:31,835 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 20:25:32,864 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1 HTTP/11" 200 453
2025-04-04 20:25:32,865 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=New+Delhi%2C+Delhi&format=json&limit=1
2025-04-04 20:25:34,068 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=New+Delhi%2C+Delhi&format=json&limit=1 HTTP/11" 200 528
2025-04-04 20:25:34,069 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 20:25:35,023 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/11" 200 None
2025-04-04 20:27:40,492 - DEBUG - connect_tcp.started host='api.phidata.com' port=443 local_address=None timeout=60 socket_options=None
2025-04-04 20:27:40,967 - DEBUG - connect_tcp.complete return_value=<httpcore._backends.sync.SyncStream object at 0x0000027834060EF0>
2025-04-04 20:27:40,967 - DEBUG - start_tls.started ssl_context=<ssl.SSLContext object at 0x00000278348AB050> server_hostname='api.phidata.com' timeout=60
2025-04-04 20:27:41,556 - DEBUG - start_tls.complete return_value=<httpcore._backends.sync.SyncStream object at 0x0000027833DF6090>
2025-04-04 20:27:41,557 - DEBUG - send_request_headers.started request=<Request [b'POST']>
2025-04-04 20:27:41,558 - DEBUG - send_request_headers.complete
2025-04-04 20:27:41,558 - DEBUG - send_request_body.started request=<Request [b'POST']>
2025-04-04 20:27:41,558 - DEBUG - send_request_body.complete
2025-04-04 20:27:41,559 - DEBUG - receive_response_headers.started request=<Request [b'POST']>
2025-04-04 20:27:41,842 - DEBUG - receive_response_headers.complete return_value=(b'HTTP/1.1', 200, b'OK', [(b'Date', b'Fri, 04 Apr 2025 14:57:42 GMT'), (b'Content-Type', b'application/json'), (b'Content-Length', b'72'), (b'Connection', b'keep-alive'), (b'server', b'uvicorn')])
2025-04-04 20:27:41,843 - INFO - HTTP Request: POST https://api.phidata.com/v1/telemetry/agent/run/create "HTTP/1.1 200 OK"
2025-04-04 20:27:41,843 - DEBUG - receive_response_body.started request=<Request [b'POST']>
2025-04-04 20:27:41,843 - DEBUG - receive_response_body.complete
2025-04-04 20:27:41,843 - DEBUG - response_closed.started
2025-04-04 20:27:41,844 - DEBUG - response_closed.complete
2025-04-04 20:27:41,844 - DEBUG - close.started
2025-04-04 20:27:41,844 - DEBUG - close.complete
2025-04-04 20:35:02,070 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\forms.py changed, reloading.
2025-04-04 20:35:49,588 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\views.py changed, reloading.
2025-04-04 20:35:56,836 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\views.py changed, reloading.
2025-04-04 20:36:33,973 - INFO - C:\Users\Ganesh\Hackathons\supplyChainGS\Supplychain\supplier\forms.py changed, reloading.
2025-04-04 20:37:36,197 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:37:36,198 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:37:36,198 - INFO - Geocoder initialized successfully
2025-04-04 20:37:36,198 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1
2025-04-04 20:37:36,199 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 20:37:36,904 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1 HTTP/11" 200 453
2025-04-04 20:37:36,905 - DEBUG - RateLimiter sleep(0.2943597999983467)
2025-04-04 20:37:37,200 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=New+Delhi%2C+Delhi&format=json&limit=1
2025-04-04 20:37:38,073 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=New+Delhi%2C+Delhi&format=json&limit=1 HTTP/11" 200 528
2025-04-04 20:37:38,075 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 20:37:38,947 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/11" 200 None
2025-04-04 20:38:23,638 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:38:23,638 - DEBUG - Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2025-04-04 20:38:23,639 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1
2025-04-04 20:38:23,639 - DEBUG - Starting new HTTPS connection (1): nominatim.openstreetmap.org:443
2025-04-04 20:38:24,260 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=Noida%2C+Uttar+Pradesh&format=json&limit=1 HTTP/11" 200 453
2025-04-04 20:38:24,260 - DEBUG - RateLimiter sleep(0.37887339999724645)
2025-04-04 20:38:24,640 - DEBUG - Nominatim.geocode: https://nominatim.openstreetmap.org/search?q=New+Delhi%2C+Delhi&format=json&limit=1
2025-04-04 20:38:25,343 - DEBUG - https://nominatim.openstreetmap.org:443 "GET /search?q=New+Delhi%2C+Delhi&format=json&limit=1 HTTP/11" 200 528
2025-04-04 20:38:25,345 - DEBUG - Starting new HTTPS connection (1): api.openrouteservice.org:443
2025-04-04 20:38:26,509 - DEBUG - https://api.openrouteservice.org:443 "POST /v2/directions/driving-car HTTP/11" 200 None
2025-04-05 10:24:57,548 - WARNING - Not Found: /favicon.ico
2025-04-05 10:24:57,560 - WARNING - Not Found: /favicon.ico
2025-04-05 10:26:31,323 - WARNING - Not Found: /favicon.ico
2025-04-05 10:35:41,155 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py changed, reloading.
2025-04-05 10:36:01,844 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py changed, reloading.
2025-04-05 10:36:10,345 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py changed, reloading.
2025-04-05 10:38:56,808 - ERROR - Internal Server Error: /supplier/profile/edit/
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'else'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py", line 271, in edit_profile
    return render(request, 'supplier/edit_profile.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 563, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 115: 'else', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
2025-04-05 10:40:50,960 - ERROR - Internal Server Error: /supplier/profile/
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'else'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py", line 234, in view_profile
    return render(request, 'supplier/profile.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 563, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 74: 'else', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
2025-04-05 10:40:56,422 - ERROR - Internal Server Error: /supplier/profile/
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'else'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py", line 234, in view_profile
    return render(request, 'supplier/profile.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 563, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 74: 'else', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
2025-04-05 10:41:16,983 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py changed, reloading.
2025-04-05 10:41:21,899 - ERROR - Internal Server Error: /supplier/profile/
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'else'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py", line 234, in view_profile
    return render(request, 'supplier/profile.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 962, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Python312\Lib\site-packages\django\template\base.py", line 563, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 74: 'else', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
2025-04-05 10:49:00,012 - INFO - C:\Users\Lenovo\Desktop\supplychain management\backendproject\Supplychain\supplier\views.py changed, reloading.
